### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by fredroy**
Based on
 - #28 

(nice to check regression for every commit)

- Replace beginEdit/endEdit with writeAcessor
- Remove using namespace and make it explicit with vector is used (std:: or type:: ???)
- Remove one useless specialized function
- Remove warnings about conversion from int to unsigned (or float??)
- Remove one weird use of indices as float (??)
- Change NULL to nullptr
- use checkEventType instead of a dyn_cast
- clean dtors
- remove useless templatename
- etc....

### CODE REVIEW

File: src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl [hunk 1]
```diff
@@ -547,7 +536,7 @@ void AdaptiveInflatableBeamForceField<DataTypes>::addForce (const MechanicalPara
         computeGravityVector();
 
     /// TODO:
-    ///* Redimentionner _localBeamMatrices
+    ///* Redimensionner _localBeamMatrices
```
Comment (epernod) [inline src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl:539]:
```suggestion
    ///* Resize _localBeamMatrices

```

File: src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl [hunk 2]
```diff
@@ -547,7 +536,7 @@ void AdaptiveInflatableBeamForceField<DataTypes>::addForce (const MechanicalPara
         computeGravityVector();
 
     /// TODO:
-    ///* Redimentionner _localBeamMatrices
+    ///* Redimensionner _localBeamMatrices
     ///* Calculer les rotation et les transformations
```
Comment (epernod) [inline src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl:540]:
```suggestion
    ///* Compute rotations and transformations

```

File: src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl [hunk 3]
```diff
@@ -547,7 +536,7 @@ void AdaptiveInflatableBeamForceField<DataTypes>::addForce (const MechanicalPara
         computeGravityVector();
 
     /// TODO:
-    ///* Redimentionner _localBeamMatrices
+    ///* Redimensionner _localBeamMatrices
     ///* Calculer les rotation et les transformations
     ///* Calculer la matrice "locale"
```
Comment (epernod) [inline src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl:541]:
```suggestion
    ///* Compute local frame matrix

```

File: src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl [hunk 4]
```diff
@@ -547,7 +536,7 @@ void AdaptiveInflatableBeamForceField<DataTypes>::addForce (const MechanicalPara
         computeGravityVector();
 
     /// TODO:
-    ///* Redimentionner _localBeamMatrices
+    ///* Redimensionner _localBeamMatrices
     ///* Calculer les rotation et les transformations
     ///* Calculer la matrice "locale"
     ///* Calculer la force exercée par chaque beam
```
Comment (epernod) [inline src/BeamAdapter/component/forcefield/AdaptiveInflatableBeamForceField.inl:542]:
```suggestion
    ///* Compute force contribution of each beam

```

File: src/BeamAdapter/component/constraint/AdaptiveBeamSlidingConstraint.inl [hunk 5]
```diff
@@ -244,7 +241,7 @@ void AdaptiveBeamSlidingConstraint<DataTypes>::getConstraintViolation(const Cons
 
 template<class DataTypes>
 void AdaptiveBeamSlidingConstraint<DataTypes>::getConstraintResolution(const ConstraintParams* cParams,
-                                                                vector<ConstraintResolution*>& resTab,
+                                                                std::vector<ConstraintResolution*>& resTab,
```
[the remark this answers is not in the document]
Comment (fredroy) [reply src/BeamAdapter/component/constraint/AdaptiveBeamSlidingConstraint.inl:244]: getConstraintResolution() overrides a method in BaseConstraint which takes.... a std::vector 🙈

File: src/BeamAdapter/component/engine/WireRestShape.inl [hunk 6]
```diff
@@ -588,32 +582,25 @@ void WireRestShape<DataTypes>::initFromLoader()
     if (!checkTopology())
         return;
 
-    vector<Vec3> vertices;
-    vector<Vec2> edges;
+    type::vector<Vec3> vertices;
+    sofa::core::topology::BaseMeshTopology::SeqEdges edges;
```
[the remark this answers is not in the document]
Comment (fredroy) [reply src/BeamAdapter/component/engine/WireRestShape.inl:586]: SeqEdges is not defined in Topology, but in BaseMeshTopology.... 🙊

File: src/BeamAdapter/component/BeamInterpolation.inl [hunk 7]
```diff
@@ -545,19 +527,13 @@ void BeamInterpolation<DataTypes>::setTransformBetweenDofAndNode(int beam, const
 
     if (!zeroORone)
     {
-        vector<Transform> &DOF0_TransformNode0 = *d_DOF0TransformNode0.beginEdit();
-
-        DOF0_TransformNode0[beam] = DOF_H_Node;
-
-        d_DOF0TransformNode0.endEdit();
+        auto DOF0TransformNode0 = sofa::helper::getWriteOnlyAccessor(d_DOF0TransformNode0);
+        DOF0TransformNode0[beam] = DOF_H_Node;
```
[the remark this answers is not in the document]
Comment (fredroy) [reply src/BeamAdapter/component/BeamInterpolation.inl:531]: DOF_H_Node is a Transform& and d_DOF0_TransformNode0 is a Data of vector of Transform.
In this case, the intent was to copy DOF_H_Node at the index `beam` in the value of the Data.
The accessor implements the operator[] so it should do its job.

And the best proof is that it compiles 🐻 (without warning of cast or anything)

**Review by epernod — APPROVED** [review]

**Pull request merged**